compiler/rustc_hir_typeck/src/_match.rs

```diff
@@ -139,7 +139,30 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 &cause,
                 Some(&arm.body),
                 arm_ty,
-                |err| self.suggest_removing_semicolon_for_coerce(err, expr, arm_ty, prior_arm),
+                |err| {
+                    if let hir::ExprKind::Block(block, _) = arm.body.kind
```

Comment:
Please hide this closure body behind a function!

Reply:
Done